hadoop-hdfs-project/hadoop-hdfs/src/test/java/org/apache/hadoop/hdfs/tools/TestDFSAdmin.java

```diff
@@ -1249,8 +1249,15 @@ public void testAllDatanodesReconfig()
     when(reconfigurationUtil.parseChangedProperties(any(Configuration.class),
         any(Configuration.class))).thenReturn(changes);
 
-    int result = admin.startReconfiguration("datanode", "livenodes");
-    assertThat(result).isEqualTo(0);
+    LambdaTestUtils.await(1000, 50, () -> {
+      int result = admin.startReconfiguration("datanode", "livenodes");
+      assertThat(result).isEqualTo(0);
+      final List<String> outs = new ArrayList<>();
+      final List<String> errs = new ArrayList<>();
+      awaitReconfigurationFinished("datanode", "livenodes", outs, errs);
+      return errs.isEmpty();
+    });
+
```

Comment:
in this logic we are doing `` admin.startReconfiguration("datanode", "livenodes");`` like in every iteration, we should just wait for the result and not perform any action, that changes the flow of the test